### PR TITLE
Add procedure context 

### PR DESCRIPTION
**Pull request by tnachen**
Addresses #106

### CODE REVIEW

File: core/src/operations/proc.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use core::fmt;
+
+/// Stores procedure information
```
Comment (bobbinth) [inline core/src/operations/proc.rs:3]: nit: could we say "Contains procedure metadata (without procedure body)"?

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! Looks good. I left some inline comments.

The biggest thing I'm still tying to think through is whether it would be possible to keep `Operation` as `Copy` as I'm not sure what could performance implications of getting rid of `Copy` be.

File: core/src/operations/proc.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#[derive(Clone, Debug, PartialEq)]
+pub struct ProcInfo {
+    name: String,
+    locals: usize,
+}
+
+impl fmt::Display for ProcInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "proc.{}.{}", self.name, self.locals)?;
```
[the remark this answers is not in the document]
Comment (tnachen) [reply core/src/operations/proc.rs:12]: ah yes, sorry I forgot about this

File: processor/src/operations/mod.rs [hunk 3]
```diff
@@ -129,11 +129,13 @@ impl Process {
             // ----- cryptographic operations -----------------------------------------------------
             Operation::RpPerm => self.op_rpperm()?,
             Operation::MpVerify => self.op_mpverify()?,
-            Operation::MrUpdate(copy) => self.op_mrupdate(copy)?,
+            Operation::MrUpdate(copy) => self.op_mrupdate(*copy)?,
 
             // ----- decorators -------------------------------------------------------------------
-            Operation::Debug(options) => self.op_debug(options)?,
-            Operation::Advice(injector) => self.op_advice(injector)?,
+            Operation::Debug(options) => self.op_debug(*options)?,
```
Comment (bobbinth) [inline processor/src/operations/mod.rs:135]: Should we pass debug options by reference here?

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! A few small comments inline.

A couple of additional things:
1. Could you rebase from the latest `next`? There was one small refactoring which may not compile with `Operation` not being `Copy`.
2. There is one piece missing: we need to modify the procedure parser to actually inserted `ProcStart` and `ProcEnd` operations into the program during parsing.

**Comment by bobbinth** [comment]
Superseded by #315

**Pull request closed without merging**